prompt_poet/template_registry.py

```diff
@@ -40,42 +41,26 @@ def __init__(
             self._initialized = True
 
     def get_template(
-        self,
-        template_name: str,
-        template_dir: str,
-        package_name: str = None,
-        use_cache: bool = False,
+            self,
+            template_loader: TemplateLoader,
+            use_cache: bool = False,
     ) -> j2.Template:
         """Get template from cache or load from disk.
 
-        :param template_name: The name of the file containing the raw template.
-        :param template_dir: The path to the directory housing the file
-            `template_name`.
-        :param package_name: An optional parameter indicating to find
-            `template_name` within `template_dir` within a python package
-            `package_name`.
-        :param use_examples: An optional parameter indicating to use the
-            examples packaged into the the Prompt Poet package.
-        :param use_cache: An optional parameter indicating to use the
-            examples packaged into the the Prompt Poet package.
-        """
-        if template_dir.endswith("/"):
-            raise ValueError(
-                f"template_dir must not end with a '/'. Found: {template_dir}"
-            )
-
-        cache_key = self._build_cache_key(
-            template_name=template_name,
-            template_dir=template_dir,
-            package_name=package_name,
-        )
+        Args:
+            template_loader: A TemplateLoader instance that handles loading the template
+                from its source.
+            use_cache: Whether to use cached template if available. If False or the
+                template is not in cache, it will be loaded from disk.
```

Comment:
nit: remain consistent with the use of :param

Reply:
done